[cmake] Fix Bluetooth/lcms2 pkgconfig search name

fuzzard opened the pull request:
## Description
Remove lcms2 pkgconfig search name. Distro's appear to have lcms2.pc
Add Bluetooth pkgconfig search name. Linux default is Bluez for bluetooth. Not relevant for any other platforms, so should be sufficient

## Motivation and context
https://github.com/xbmc/xbmc/pull/26847#issuecomment-3513541507

## How has this been tested?
N/A

## What is the effect on users?
Linux users can use ENABLE_BLUETOOTH and lcms2 should be found.

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` with no space in all the boxes that apply like this: [X] -->
- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` with no space in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

Commit:
[cmake] FindLCMS2 remove custom pkgconfig search name

Fedora/Debian supply lcms2.pc so remove custom search name of liblcms2

fuzzard commented:
You're going to have to start providing logs and info at this point.

mackal commented:
> You're going to have to start providing logs and info at this point.

When you build with `-DENABLE_BLUETOOTH=ON` it's failing because on the line I linked above it's using the CMake functions (not kodi helper stuff) to search for the modules bluez and bluetooth. This fails because passing `ON` makes it required. (`AUTO` doesn't fail because 1 error is fine) I'm not sure why the cmake is checking for both. Looking at BlueZ source code I think they only install `bluez.pc`


```
-- MODULE: BLUETOOTH
-- bluetooth_LIB_TYPE: target
-- DEP_LOCATION: /var/tmp/portage/media-tv/kodi-9999/work/kodi-9999_build/build
-- PROJECTSOURCE: /var/tmp/portage/media-tv/kodi-9999/work/kodi-9999
-- Checking for modules 'bluez;bluetooth'
--   Package 'bluetooth' not found
```

CrossGUID issue I'm not 100% sure on

```
-- MODULE: CROSSGUID
-- crossguid_LIB_TYPE: target
-- DEP_LOCATION: /var/tmp/portage/media-tv/kodi-9999/work/kodi-9999_build/build
-- PROJECTSOURCE: /var/tmp/portage/media-tv/kodi-9999/work/kodi-9999
```

I have both pkgconfig and cmake stuff installed:

```
/usr/lib64/pkgconfig/crossguid.pc
/usr/share/crossguid/cmake/crossguid-config-relwithdebinfo.cmake
/usr/share/crossguid/cmake/crossguid-config.cmake
```

mackal commented:
> The bluetooth line you pointed to is a pkgconfig search, not a cmake config search.
> 
> Show the paths of the pc file for it.
> 
> I would have a guess that the paths you provided for crossguid arent in the "standard" search paths cmake uses. Might explain the bluez failure as well.
> 
> We only provide HINTS paths that seem to be "common". This only effects a cmake config search (ie crossguid in the info you have provided)
> 
> https://github.com/xbmc/xbmc/blob/77c3c503eed6ae200a1e7e22718c3efd32fcff21/cmake/scripts/common/ModuleHelpers.cmake#L701-L704
> 
> Im guessing theres going to be some env var that you may be able to provide to tell pkgconfig to search in the lib64 folder (and not lib). I cant see any way i can influence that at a 20 second glance in cmake.

https://cmake.org/cmake/help/latest/command/find_package.html#search-procedure If I understand this correctly, those should be standard search paths.

Commit:
[cmake] FindBluetooth remove secondary search name

The second search name causes the expected variables to not be set. Bluez is considered
the defacto for unix, so just drop back to a single search for it.

mackal commented:
Bluetooth issues resolved now.

heitbaum commented:
I have to revert this to build with 5232dc14b74842a24830dad07b12be4ded33b055
```
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::Deinitialize():(.text+0x4acc): error: undefined reference to 'sdp_close'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x7888): error: undefined reference to 'sdp_record_alloc'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x789a): error: undefined reference to 'sdp_uuid128_create'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x78bc): error: undefined reference to 'sdp_set_service_id'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x78d2): error: undefined reference to 'sdp_uuid16_create'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x78e1): error: undefined reference to 'sdp_list_append'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x78f6): error: undefined reference to 'sdp_set_uuidseq_attr'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x7908): error: undefined reference to 'sdp_uuid16_create'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x7917): error: undefined reference to 'sdp_list_append'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x7926): error: undefined reference to 'sdp_list_append'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x7938): error: undefined reference to 'sdp_uuid16_create'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x7947): error: undefined reference to 'sdp_data_alloc'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x7956): error: undefined reference to 'sdp_list_append'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x7989): error: undefined reference to 'sdp_set_access_protos'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x79a0): error: undefined reference to 'sdp_set_info_attr'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x79c1): error: undefined reference to 'sdp_set_uuidseq_attr'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x79c9): error: undefined reference to 'sdp_data_free'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x79d5): error: undefined reference to 'sdp_list_free'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x79df): error: undefined reference to 'sdp_list_free'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x79eb): error: undefined reference to 'sdp_list_free'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x79f7): error: undefined reference to 'sdp_list_free'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x7a17): error: undefined reference to 'sdp_connect'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x7a32): error: undefined reference to 'sdp_record_register'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x7ba7): error: undefined reference to 'sdp_record_free'
/tmp/ccSSdNfQ.ltrans310.ltrans.o:<artificial>:function JSONRPC::CTCPServer::InitializeBlue():(.text+0x7c0e): error: undefined reference to 'sdp_close'
collect2: error: ld returned 1 exit status
```
